### ISSUE TITLE
Run tests with output paths in a temporary directory instead of project-local `spooned` directory

### ISSUE THREAD

**Issue by slarse** [INRIA/spoon#4677]
As pointed out in #4672, the `spooned` directory persists across tests and causes interdependencies between tests. It can sometimes be incredibly confusing and create very hard to debug errors (until you realize you need to remove the `spooned` directory and everything will be fine).

All tests should execute in a temporary directory so as not to affect other tests. Tests that depend on certain files and directories existing should create those files and directories. Fixing this will probably be a somewhat lengthy process as there are many tests to check and update.

The definition of done for this issue is that no test creates the `spooned` directory in the current working directory. There are some tests that generate code for Spoon, for which the output is then copy-pasted into the production code. We need to still be able to get that code somehow (but not necessarily by running the test suite).

**Comment by SirYwell** [comment]
We also noticed that a lot of tests are generating files in `target/`. While those are at least removed with `mvn clean`, I think depending on actual temporary directories would make sense there too.

**Comment by slarse** [comment]
Yes, temporary directories wherever possible is preferable as it decouples the tests from each other.
